src/runner.js

```diff
@@ -1,40 +1,57 @@
 // @flow
 import moment from 'moment';
+import kafka from 'no-kafka';
 import KafkaClient from './helpers/kafka';
 
-import type { Env } from '../types';
+import type { Config } from '../types';
 
-const Runner = (env: Env, registry: Object, logger: Object) => {
+const Runner = (config: Config, registry: Object, logger: Object) => {
   const kafkaClient = KafkaClient({
-    kafkaConnection: env.KAFKA_CONNECTION,
-    kafkaCodec: env.KAFKA_CODEC,
-    clientId: env.CLIENT_ID,
+    kafkaConnection: config.kafkaConnection,
+    kafkaCodec: config.kafkaCodec,
+    clientId: config.clientId,
     logger: {
-      logLevel: env.LOG_LEVEL,
+      logLevel: config.logLevel,
     },
-    kafkaGroupId: env.KAFKA_GROUP_ID,
+    kafkaGroupId: config.kafkaGroupId,
   });
 
-
-  const receive = async (payload: Object, topic: string) => {
-    logger.info('Payload: ', JSON.stringify(payload, null, 2), 'received on topic:', topic);
-    const task = registry[topic];
-    try {
-      await task.subscribe(payload);
-    } catch (ex) {
-      logger.error('Error while executing consumer callback ', ex);
-    }
-  };
-
+  const receive = (messageSet: Array<Object>, topic: string, partition: number) =>
+    Promise.all(messageSet.map(async (m) => {
+      try {
+        logger.info(`
+        *****CONSUME********
+        topic:- ${topic}
+        ********************
+        payload:- ${m.message.value}
+        ********************
+      `);
+        // commit offset
+        await kafkaClient.consumer.commitOffset({ topic, partition, offset: m.offset, metadata: 'optional' });
+        const task = registry[topic];
+        await task.subscribe(JSON.parse(m.message.value.toString('utf8')));
+      } catch (ex) {
+        logger.error('Error while executing consumer callback ', ex);
+      }
+    }));
 
   const initializeConsumer = (subscriptions: Array<string>) => {
     logger.info('initializing consumer', subscriptions);
+
     return kafkaClient.consumer.init([{
```

Comment:
can we use dependency injection for the consumer, rather than depending on the kafkaHelper file. will allow library users to pick their strategy